editor-service/src/main/java/com/qreal/robots/database/users/client/UserServiceImpl.java

```diff
@@ -0,0 +1,107 @@
+package com.qreal.robots.database.users.client;
+
+import com.qreal.robots.database.users.model.User;
+import com.qreal.robots.thrift.gen.TUser;
+import com.qreal.robots.thrift.gen.UserDbService;
+import org.apache.thrift.TException;
+import org.apache.thrift.protocol.TBinaryProtocol;
+import org.apache.thrift.protocol.TProtocol;
+import org.apache.thrift.transport.TSocket;
+import org.apache.thrift.transport.TTransport;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+
+/**
+ * Thrift client side of UserDBService.
+ */
+@Service("userService")
+public class UserServiceImpl implements UserService {
+
+    private static final Logger logger = LoggerFactory.getLogger(UserServiceImpl.class);
+
+    private TTransport transport;
+
+    private UserDbService.Client client;
+
+    /**
+     * Constructor creates connection with Thrift TServer.
+     */
+    public UserServiceImpl() {
+        String url = "localhost";
+        int port = 9090;
+        logger.info("Client UserService was created with Thrift socket on url = {}, port = {}", url, port);
+        transport = new TSocket(url, port);
+        TProtocol protocol = new TBinaryProtocol(transport);
+        client = new UserDbService.Client(protocol);
+    }
+
+    @Override
+    @Transactional
+    public void save(User user) {
+        logger.trace("save method called with parameters: user = {}", user.getUsername());
+        try {
+            transport.open();
```

Comment:
we could extract a function here that opens the transport, performs an action provided as an argument and closes the transport